src/Catty/DataModel/UserData/UserVariable.swift

```diff
@@ -59,4 +60,17 @@
         return self
     }
 
+    func set(_ value: Any?) {
```

Comment:
This custom setter has the downside that the "content" of `value` can still be set using the default setter for the variable. Please override the default setter, so that your code always get executed when calling `userVariable.value = value`. You might want to change `value` to a computed property and introduce a new private property (e.g., `_value`). In this setter please also include all the logic from the old Objective C setter about `NSString` and `NSNumber`. Unfortunately the tests have not been strong enough so far, so please add tests for those execution paths into the new `UserVariableTests` as well. Thanks!

Reply:
PS: if you find another way to override the setter, this will also be fine. Most important: it should not be able to set a `value` without your code being executed.